app/src/main/java/com/boosters/promise/ui/detail/PromiseDetailActivity.kt

```diff
@@ -0,0 +1,89 @@
+package com.boosters.promise.ui.detail
+
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.annotation.UiThread
+import androidx.databinding.DataBindingUtil
+import com.boosters.promise.R
+import com.boosters.promise.data.model.Location
+import com.boosters.promise.databinding.ActivityPromiseDetailBinding
+import com.boosters.promise.ui.detail.adapter.PromiseMemberAdapter
+import com.boosters.promise.ui.promisesetting.model.PromiseUiState
+import com.naver.maps.geometry.LatLng
+import com.naver.maps.geometry.Tm128
+import com.naver.maps.map.CameraUpdate
+import com.naver.maps.map.MapFragment
+import com.naver.maps.map.NaverMap
+import com.naver.maps.map.OnMapReadyCallback
+import com.naver.maps.map.overlay.Marker
+
+class PromiseDetailActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    private lateinit var binding: ActivityPromiseDetailBinding
+    private val promiseDetailViewModel: PromiseDetailViewModel by viewModels()
+    private val promiseMemberAdapter = PromiseMemberAdapter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_promise_detail)
+        setBinding()
+        setPromiseInfo()
+        initMap()
+
+        promiseMemberAdapter.submitList(promiseDetailViewModel.promiseInfo.value?.members)
+    }
+
+    @UiThread
+    override fun onMapReady(map: NaverMap) {
+        val location = requireNotNull(promiseDetailViewModel.promiseInfo.value).destinationLocation
+        val destinationLocation = convertLocation(location)
```

Comment:
nullable이 아닌거같은데 requireNotNull을 사용하신 이유가 있나요?

Reply:
LiveData인 promiseInfo 변수가 nullable로 인식돼서 추가해줬습니다.
```suggestion
    override fun onMapReady(map: NaverMap) {
        val location = promiseDetailViewModel.promiseInfo.value?.destinationLocation
        if (location != null) {
            val destinationLocation = convertLocation(location)
```
사용하지 않는다면 이렇게 변경될 수 있을 것같습니다!

Reply:
@jerrytrap 
아마 livedata를 옵저브하는게 아니라 직접 liveData.value로 가져오다보니 그런거같아요.
직접 livedata값을 가져오는것보다 옵저브하는게 좋아보여요

Reply:
옵저브해서 가져오려면 어떻게 해야하나요? 방법이 잘 떠오르지 않네요

Reply:
@jerrytrap 
promiseInfo를 필요로하는 로직을 옵저브 안에 구현해야하지 않을까싶네요,